modules/layouts/LayoutCS.chpl

```diff
@@ -255,10 +257,21 @@ class CSDom: BaseSparseDomImpl {
     const (row, col) = ind;
 
     var ret: (bool, idxType);
-    if this.compressRows then
-      ret = binarySearch(idx, col, lo=startIdx(row), hi=stopIdx(row));
-    else
-      ret = binarySearch(idx, row, lo=startIdx(col), hi=stopIdx(col));
+    if this.compressRows {
+      if this.sorted then
+        ret = binarySearch(idx, col, lo=startIdx(row), hi=stopIdx(row));
+      else {
+        ret = linearSearch(idx, col, lo=startIdx(row), hi=stopIdx(row));
+        if !ret[1] then ret[2] = stopIdx(row)+1;
```

Comment:
If linearSearch() returned `hi+1` rather than `Dom.high+1`, would the extra conditional required for linearSearch() over binarySearch() no longer be required?  (for the purpose of symmetry; also because it seems a shame that that routine takes that approach).

Reply:
Yes that would be nice. I think that would also fit well with how BinarySearch works.

Reply:
I've made this it's own issue (#9908).
Also, instead of hi+1, I'm going with hi+stride (current behavior is Dom.high+stride)